### ISSUE TITLE
[bug] objective/entropy < 0  when using rlootrainer and ppotrainer

### ISSUE THREAD

**Issue by macheng6** [huggingface/trl#2496]
https://github.com/huggingface/trl/blob/1661bc295e694dc1ec9d50f80746612347051da3/trl/trainer/rloo_trainer.py#L443

This is because in the previous code, the padding part of logprod is filled with 1.

INVALID_LOGPROB=1.0
logprobs = torch.masked_fill(logprobs, padding_mask, INVALID_LOGPROB)

I don't know why INVALID_LOGPROB is set to 1, wouldn't it work fine if it is set to 0?

**Comment by asparius** [comment]
This has been noted previously #2281. I believe this was introduced in PPOv2 which was replication of the openai tldr paper which also contains this INVALID_LOGPROB=1.0 which does not break training because it cancels out at kl reward. Perhaps @vwxyzjn can tell why this was used, instead of masked_mean version

**Comment by albertvillanova** [comment]
Hey @macheng6, sorry for the very slow reply here — we're catching up on the backlog. 

You were right, and both halves of this are now fixed on main. 

For PPO (which now lives at `trl.experimental.ppo`), #6121 masks the padding positions out of the metric, so `objective/entropy` can no longer be dragged negative by the `INVALID_LOGPROB` sentinel: https://github.com/huggingface/trl/blob/38b4b810c202b43d761eaa43ad160b6bf3fc65f1/trl/experimental/ppo/ppo_trainer.py#L883-L884. 

For RLOO, the trainer was fully rewritten in #3801 — `INVALID_LOGPROB` is gone entirely and entropy is now computed from the logits and normalized by the completion mask, logged as `entropy` rather than `objective/entropy`: https://github.com/huggingface/trl/blob/38b4b810c202b43d761eaa43ad160b6bf3fc65f1/trl/trainer/rloo_trainer.py#L1727-L1730. 

To your question about the sentinel value: 1.0 is deliberate — a real log-prob is always <= 0, so 1.0 is an impossible value that makes masked positions unmistakable, whereas 0.0 is a perfectly legal log-prob (p=1) and would silently blend in. The bug was that the entropy metric forgot to apply the mask, not the sentinel itself. 

Closing as fixed — the PPO fix isn't in v1.9.2 yet, so you'll need main or the next release.
